pkg/dynamicinstrumentation/di.go

```diff
@@ -80,9 +80,7 @@ func RunDynamicInstrumentation(opts *DIOptions) (*GoDI, error) {
 	if err != nil {
 		return nil, err
 	}
-	stopFunctions := []func(){
```

Comment:
Just removing the StopGlobalDiagnostics function which was breaking tests.

Reply:
That function closes the diagnostic upload channel at the moment - I guess we should have other cleanup for other resources here?

Right now I don't think removing it makes much difference since the lifetime of Go DI is tied to the system-probe, except in the case of tests but it's surprising that this makes tests fail.

Eventually we will need to support turning DI on / off independently of the lifecycle of the system probe so cleanup will be necessary.